File: leetcode/c++/2 Add Two Numbers.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *head = new ListNode(-2);
        ListNode *index = head;
        int carry = 0;
        
        while(l1 || l2){
            int val1 = l1 ? l1 -> val : 0;
            int val2 = l2 ? l2 -> val : 0;
            int sum = val1 + val2 + carry;
            carry = sum / 10;
            sum %= 10;
            index -> next = new ListNode(sum);
            index = index -> next;
            if(l1){
                l1 = l1 -> next;
            }
            if(l2){
                l2 = l2 -> next;
            }
        }
        if(carry){
            index -> next = new ListNode(carry);
        }
        return head -> next;
    }
};

File: leetcode/ts/2 Add Two Numbers.ts
function addTwoNumbers(l1: ListNode | null, l2: ListNode | null): ListNode | null {
    let carry : number = 0
    let dummy : ListNode = new ListNode(-1, null)
    let head : ListNode = dummy
    
    while(l1 != null || l2 != null || carry != 0){
        let num1 : number = l1 ? l1.val : 0
        let num2 : number = l2 ? l2.val : 0
        let sum : number = num1 + num2 + carry
        carry = Math.floor(sum / 10)
        head.next = new ListNode(sum % 10)
        head = head.next
        l1 ? l1 = l1.next : null
        l2 ? l2 = l2.next : null
    }
    return dummy.next
};